src/org/labkey/cds/CDSController.java

```diff
@@ -1041,18 +1041,22 @@ else if (isPost())
         @Override
         public Object execute(StudyDocumentForm form, BindException errors)
         {
-            if (null == form.getFilename())
+            if (StringUtils.isBlank(form.getFilename()))
             {
+                errors.reject(ERROR_MSG, "filename parameter required");
                 return false;
             }
 
             boolean isValidLink = false;
             WebdavService service = ServiceRegistry.get().getService(WebdavService.class);
-            WebdavResource resource = service.lookup(Path.parse(form.filename));
+            LOG.info("File path: " + Path.parse(form.getFilename()).toString()); //TODO: Added for investigating Ticket 40760, to be removed.
+            WebdavResource resource = service.lookup(Path.parse(form.getFilename()));
 
             if (resource != null)
             {
                 File requestedFile = resource.getFile();
+                LOG.info(requestedFile.getName() + " is present: " + (requestedFile != null));//TODO: Added for investigating Ticket 40760, to be removed.
+                LOG.info(requestedFile.getName() + " is readable: " + requestedFile.canRead());//TODO: Added for investigating Ticket 40760, to be removed.
```

Comment:
if requestedFile is null, this would throw NPE.

Reply:
yes, good point, now updated.